configuration/azure/appconfig/appconfig.go

```diff
@@ -248,9 +266,103 @@ func (r *ConfigurationStore) getLabelFromMetadata(metadata map[string]string) *s
 }
 
 func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
-	return "", fmt.Errorf("Subscribe is not implemented by this configuration store")
+	subscribeID := uuid.New().String()
+	stop := make(chan struct{})
+	r.subscribeStopChanMap.Store(subscribeID, stop)
+	sentinelKey := r.getSentinelKeyFromMetadata(req.Metadata)
+	if sentinelKey == "" {
+		return "", fmt.Errorf("sentinel key is not provided in metadata")
+	}
+	actualKeyToSubscribeSlice := make([]string, 0)
+	for _, key := range req.Keys {
+		_, found := r.keysToSubIDMap[key]
```

Comment:
This was intentional actually. Since that sentinel key is already being watched, so any change to that sentinel key will fetch the new values. So there is no need to watch again for same key in a diff subscribe call.